### PR TITLE
Add types to task.py

### PR DESCRIPTION
**Pull request by InvincibleRMC**

### CODE REVIEW

File: rclpy/rclpy/task.py [hunk 1]
```diff
@@ -128,13 +126,14 @@ def set_exception(self, exception):
...
 
-    def _schedule_or_invoke_done_callbacks(self):
+    def _schedule_or_invoke_done_callbacks(self) -> None:
         """
         Schedule done callbacks on the executor if possible, else run them directly.
 
         This function assumes self._lock is not held.
         """
         with self._lock:
+            assert self._executor is not None
```
Comment (fujitatomoya) [inline rclpy/rclpy/task.py:136]: this check is required here? i think this could be `None`.

**Review by fujitatomoya — COMMENTED** [review]
i have one comment to confirm. i would like to request another review.

File: rclpy/rclpy/task.py (hunk 1 above)
Comment (InvincibleRMC) [reply rclpy/rclpy/task.py:136]: I believe if `self._executor` is `None` then an error of  `'NoneType' object is not callable` would occur with the line `executor = self._executor()`

**Review by clalancette — APPROVED** [review]
This looks good to me.  I'm going to go ahead and rebase it onto the latest, since it will certainly need that in order to pass CI.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21571/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15860/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1069/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22291/)

**Pull request merged**